gnovm/pkg/gnolang/realm.go

```diff
@@ -367,6 +367,11 @@ func (rlm *Realm) FinalizeRealmTransaction(store Store) {
 	// given created and updated objects,
 	// mark all owned-ancestors also as dirty.
 	rlm.markDirtyAncestors(store)
+	// dirty object may have new real children,
+	// e.g. Map[oo1] = oo2, where oo1 as mapkey.
+	// it can be new real, but not in .newCreated,
+	// as it's not explicitly assigned via "=".
+	rlm.markNewRealChildren(store)
```

Comment:
question: I’m wondering whether we should generalize this logic, but I’m concerned that a fully generic approach could hurt performance. To date, we’ve only have the issue with mapValue, so it may be worth keeping a specialized path for that case ?

Reply:
Agreed; could we not trigger a DidUpdate hook in `PointerValue.Assign2` for map keys, in case pv.T is a map?

Reply:
yes we could. that was my first attempt...  made a separate PR here:
https://github.com/gnolang/gno/pull/4385

do you think it's better? @omarsy @thehowl

Reply:
yeah the other seems better but i wonder if its worth trying to improve the codebase so it doesn't do duplicate work. trying on the other pr...